lib/teleterm/clusters/cluster_gateways.go

```diff
@@ -42,13 +37,13 @@ type CreateGatewayParams struct {
 }
 
 // CreateGateway creates a gateway
-func (c *Cluster) CreateGateway(ctx context.Context, params CreateGatewayParams) (*gateway.Gateway, error) {
+func (c *Cluster) CreateGateway(ctx context.Context, cliCommandProvider DbcmdCLICommandProvider, params CreateGatewayParams) (*gateway.Gateway, error) {
```

Comment:
I think we should include the `DbcmdCLICommandProvider` in the `CreateGatewayParams` struct :thinking: 
Or is it a "special" param?

Reply:
Another question
The type is `DbcmdCLICommandProvider` but the variable is `cliCommandProvider` and it feels weird
One of them seems to be generic, while the other one is specific for `DB`s?
(I'm assuming this is a _provider for DB commands using the CLI_)